engine/class_modules/sc_death_knight.cpp

```diff
@@ -9295,49 +9287,52 @@ void death_knight_t::default_apl_blood()
-  standard -> add_action( this, "Marrowrend", "if=buff.bone_shield.stack<5&runic_power.deficit>=15" );
-  standard -> add_talent( this, "Bonestorm", "if=runic_power>=100&!buff.dancing_rune_weapon.up" );
-  standard -> add_action( this, "Death Strike", "if=runic_power.deficit<=(15+buff.dancing_rune_weapon.up*5+spell_targets.heart_strike*talent.heartbreaker.enabled*2)|target.1.time_to_die<10" );
-  standard -> add_action( this, "Death and Decay", "if=spell_targets.death_and_decay>=3" );
-  standard -> add_action( this, "Heart Strike", "if=buff.dancing_rune_weapon.up|rune.time_to_4<gcd" );
-  standard -> add_action( this, "Blood Boil", "if=buff.dancing_rune_weapon.up" );
-  standard -> add_action( this, "Death and Decay", "if=buff.crimson_scourge.up|talent.rapid_decomposition.enabled|spell_targets.death_and_decay>=2" );
-  standard -> add_talent( this, "Consumption" );
-  standard -> add_action( this, "Blood Boil" );
-  standard -> add_action( this, "Heart Strike", "if=rune.time_to_3<gcd|buff.bone_shield.stack>6" );
-  standard -> add_action( "use_item,name=grongs_primal_rage" ); // Because this prevents all casting it should be used during downtime
+  
+  // Night fae
+  covenants -> add_action( "death_strike,if=covenant.night_fae&buff.deaths_due.remains>3&runic_power>70", "Burn RP if we have time between DD refreshes" );
+  covenants -> add_action( "heart_strike,if=covenant.night_fae&death_and_decay.ticking&((buff.deaths_due.up|buff.dancing_rune_weapon.up)&buff.deaths_due.remains<5)", "Make sure we never lose that buff" );
+  covenants -> add_action( "deaths_due,if=!buff.deaths_due.up|buff.deaths_due.remains<3|buff.crimson_scourge.up", "And that we always cast DD as high prio when we actually need it" );
+
+  // Venthyr
+  covenants -> add_action( "death_strike,if=covenant.venthyr&runic_power>70&cooldown.swarming_mist.remains<3", "Burn RP off just before swarming comes back off CD");
+  covenants -> add_action( "swarming_mist,if=cooldown.dancing_rune_weapon.remains<3|!buff.dancing_rune_weapon.up", "And swarming as long as we're not < 3s off DRW" );
+
+  // Necrolord (forces a marrowrend on pull)
+  covenants -> add_action( "marrowrend,if=covenant.necrolord&buff.bone_shield.stack<=0", "Pre-AL marrow on pull in order to guarantee ossuary during the first DRW");
+  covenants -> add_action( "abomination_limb,if=cooldown.dancing_rune_weapon.remains<3|!buff.dancing_rune_weapon.up", "And we cast AL" );
+
+  // Kyrian
+  covenants -> add_action( "shackle_the_unworthy,if=cooldown.dancing_rune_weapon.remains<3|!buff.dancing_rune_weapon.up", "We just don't cast this during DRW" );
+
+  standard -> add_action( "blood_tap,if=rune<=2&rune.time_to_4>gcd&charges_fractional>=1.8", "Use blood tap to prevent overcapping charges if we have space for a rune and a GCD to spare to burn it" );
+  standard -> add_action( "dancing_rune_weapon,if=!talent.blooddrinker.enabled|!cooldown.blooddrinker.ready" );
```

Comment:
```suggestion
  standard -> add_action( "dancing_rune_weapon,if=!talent.blooddrinker.enabled|!cooldown.blooddrinker.ready" );
  standard -> add_action( this, "Raise Dead", "if=buff.dancing_rune_weapon.up" );
```

Ghoul adds about 7% dps to my profile

Reply:
Does it require DRW to be up? Raise dead just seems like a fire and forget and use on cooldown type of ability.

Reply:
Fixed on my branch and removed this requirement. It didn't.

Reply:
https://github.com/simulationcraft/simc/pull/5515/files#diff-2e0b4958f10e8a96eb62fa730babd633896546c4e5782ca94dec52f1d9df0d24R9292

Moved to the top since off GCD allows it, and I removed all conditionals on it since, once again, GCD allows it.

The hard part was getting sac pact to properly pop at sensible times, which is done at https://github.com/simulationcraft/simc/pull/5515/files#diff-2e0b4958f10e8a96eb62fa730babd633896546c4e5782ca94dec52f1d9df0d24R9304